doc/src/atomvm-tooling.md

```diff
@@ -0,0 +1,239 @@
+
+See the [`atomvm_rebar3_plugin`](https://github.com/atomvm/atomvm_rebar3_plugin) page for more detailed instructions about how to use the `esp32_flash` target.
+
+You can now use a serial console program such as [minicom](https://en.wikipedia.org/wiki/Minicom) or [screen](https://en.wikipedia.org/wiki/GNU_Screen) to view console output from a device.
+
+For example:
+
+        ###########################################################
+
+           ###    ########  #######  ##     ## ##     ## ##     ##
+          ## ##      ##    ##     ## ###   ### ##     ## ###   ###
+         ##   ##     ##    ##     ## #### #### ##     ## #### ####
+        ##     ##    ##    ##     ## ## ### ## ##     ## ## ### ##
+        #########    ##    ##     ## ##     ##  ##   ##  ##     ##
+        ##     ##    ##    ##     ## ##     ##   ## ##   ##     ##
+        ##     ##    ##     #######  ##     ##    ###    ##     ##
+
+        ###########################################################
+
+    I (852) AtomVM: Starting AtomVM revision 0.6.0-alpha.1
+    I (862) sys: Loaded BEAM partition boot.avm at address 0x1d0000 (size=262144 bytes)
+    I (882) network_driver: Initialized network interface
+    I (882) network_driver: Created default event loop
+    I (902) AtomVM: Found startup beam esp32init.beam
+    W (902) sys: AVM partition not found for lib.avm
+    I (902) AtomVM: Unable to mount lib.avm partition. Hopefully the AtomVM core libraries are included in your application.
+    I (922) AtomVM: Starting esp32init.beam...
+    ---
+    AtomVM init.
+    I (932) sys: Loaded BEAM partition main.avm at address 0x210000 (size=1048576 bytes)
+    Starting application...
+    Hello World
+    AtomVM finished with return value: ok
+    I (972) AtomVM: AtomVM application terminated.  Going to sleep forever ...
+
+#### STM32
+
+> TODO
+
+#### Raspberry Pico
+
+> TODO
+
+## `ExAtomVM`
+
+The [`ExAtomVM`](https://github.com/atomvm/ExAtomVM) tool is a [mix](https://elixir-lang.org/getting-started/mix-otp/introduction-to-mix.html) plugin that can be used to create and flash [Elixir](https://elixir-lang.org) applications that run over AtomVM.  Using this plugin greatly simplies the process of building Erlang applications that run over AtomVM, and is strongly encouraged for new users.
+
+### Prerequisites
+
+To use the [`ExAtomVM`](https://github.com/atomvm/ExAtomVM) tool, you will need the following software on your development machine:
+
+* A suitable version of the [Erlang/OTP](https://www.erlang.org) distribution.  See the [Release Notes](release-notes.md) for information about supported Erlang/OTP versions.
+* A suitable version of the [Elixir](https://elixir-lang.org) distribution.  See the [Release Notes](release-notes.md) for information about supported Elixir versions.
+* (optional) The [git](https://git-scm.com) command line tool, to follow examples in this chapter.
```

Comment:
At this point should git be a requirement full stop? To use the plugin users should also point to the git repository in their own applications, at least until there is a CI workflow that publishes the plugin on `hex`.